app/controllers/YourContactDetailsController.scala

```diff
@@ -42,6 +44,29 @@ class YourContactDetailsController @Inject()(authenticate: IdentifierAction,
   val log: LoggerLike = Logger(this.getClass)
 
   def onPageLoad(): Action[AnyContent] = authenticate async { implicit request =>
+
+    val localSessionId: SessionId = getSessionId()
+
+    localSessionId match {
+      case a => sessionCacheConnector.getSessionId(localSessionId.value) match {
+        case sessionCache if a.value == localSessionId.value => generateView(request, localSessionId)
+        case _ => redirectToHomePage()
+      }
+    }
+  }
+
+  def redirectToHomePage(): Future[Result] = {
+    Future.successful(Redirect(routes.CustomsFinancialsHomeController.index.url))
+  }
+
+  def getSessionId()(implicit hc: HeaderCarrier): SessionId = {
+    SessionId(hc.sessionId.getOrElse(
+      {log.error("Missing SessionID"); SessionId("Missing Session ID")}).value)
+  }
+
+  private def generateView(request: AuthenticatedRequest[AnyContent], localSessionId: SessionId)
```

Comment:
Better to move this method just after onPageLoad method as it is the second most important method after onPageLoad and it will be visited most of the times